### PR TITLE
Polling docs

### PR DESCRIPTION
**Pull request by wxtim**
This is a small change with no associated Issue. 

### Description of issue

We want to continue to allow users to over-ride platform settings of 
- `submission retry delays`
- `submission polling intervals`
- `execution polling intervals`

on a per-task basis. We are not planning to deprecate the task config items at Cylc 9.

### Description of changes:
- [x] Applied @dpmatthews 's changes to the descriptions of the three config items in `globalcfg.py`
- [x] Removed deprecation warning for the items in `workflow.py`
- [x] Import @dpmatthews descriptions of the config items from `globalcfg.py` to `workflow.py` and used these in the config description there, adding only that these over-ride the setting in `global.cylc.`.
- [x] Add a link to the list of built in job runners from the `job runner` config item.

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to both `setup.cfg` and `conda-environment.yml`.
- [x] Testing
  - [x] Already covered by existing tests.
  - [x] Built and checked Cylc Doc against these changes.
- [x] No change log entry required: This is a documentation and config deprecation warning change.
- [x] No documentation update required.

### CODE REVIEW

File: cylc/flow/cfgspec/globalcfg.py [hunk 1]
```diff
@@ -971,59 +972,84 @@
...
 
-                   5*PT1M, 10*PT5M
+                   5*PT2M, PT5M
+
+                Note that if the polling
+                :cylc:conf:`global.cylc[platforms][<platform name>]
+                communication method`
+                is used then Cylc relies on the polling to detect when a task
+                starts running so you are likely to want to configure more
+                frequent polling.
```
Comment (hjoliver) [inline cylc/flow/cfgspec/globalcfg.py:997]:
```suggestion
                Note that if the polling
                :cylc:conf:`global.cylc[platforms][<platform name>]
                communication method`
                is used then Cylc relies on polling to detect all task state changes,
                so you may want to configure more
                frequent polling.
```

File: cylc/flow/cfgspec/globalcfg.py [hunk 2]
```diff
@@ -971,59 +972,84 @@
...
             ''')
-            Conf('submission retry delays', VDR.V_INTERVAL_LIST, None, desc=f'''
-            {MOVEDFROMJOB}
+            Conf('submission retry delays', VDR.V_INTERVAL_LIST, None,
+                 desc='''
+                Cylc can automate resubmission of a task job which has not
+                been submitted sucessfully.
```
Comment (hjoliver) [inline cylc/flow/cfgspec/globalcfg.py:1002]:
```suggestion
                Cylc can automatically resubmit jobs after submission failures.
```
(Note also we agreed to retire the term "task job" in favour of just "job").

File: cylc/flow/cfgspec/globalcfg.py [hunk 3]
```diff
@@ -971,59 +972,84 @@
...
+
+                Execution retry delays are a list of ISO 8601
+                durations/intervals which tell Cylc how long to wait before
+                resubmitting a failed job.
```
Comment (hjoliver) [inline cylc/flow/cfgspec/globalcfg.py:1006]:
```suggestion
                This setting is a list of ISO 8601
                durations/intervals, for how long to wait before
                resubmitting failed jobs.
```

File: cylc/flow/cfgspec/globalcfg.py [hunk 4]
```diff
@@ -971,59 +972,84 @@
...
+
+                Each time Cylc resubmits a task job it will increment the
+                variable ``$CYLC_TASK_TRY_NUMBER`` in the task execution
+                environment. ``$CYLC_TASK_TRY_NUMBER`` allows you to vary task
+                behavior between submission attempts.
```
Comment (hjoliver) [inline cylc/flow/cfgspec/globalcfg.py:1011]:
```suggestion
                Each time Cylc resubmits a job it increments the
                variable ``$CYLC_TASK_TRY_NUMBER`` in the job execution
                environment. This allows you to vary task
                behavior between submission attempts.
```

File: cylc/flow/cfgspec/globalcfg.py [hunk 5]
```diff
@@ -971,59 +972,84 @@
...
                 Cylc can poll running jobs to catch problems that prevent task
                 messages from being sent back to the workflow, such as hard job
-                kills, network outages, or unplanned task host shutdown.
-                Routine polling is done only for the polling *task
-                communication method* (below) unless polling is
-                configured in the workflow configuration.  A list of interval
-                values can be specified, with the last value used repeatedly
-                until the task is finished - this allows more frequent polling
-                near the beginning and end of the anticipated task run time.
+                kills, network outages, or unplanned task host shutdown. A list
```
Comment (hjoliver) [inline cylc/flow/cfgspec/globalcfg.py:1021]:
```suggestion
                kills, network outages, or unplanned job host shutdown. A list
```

File: cylc/flow/cfgspec/globalcfg.py [hunk 6]
```diff
@@ -971,59 +972,84 @@
...
 
-                   5*PT1M, 10*PT5M
+                   5*PT2M, PT5M
+
+                Note that if the polling
+                :cylc:conf:`global.cylc[platforms][<platform name>]
+                communication method`
+                is used then Cylc relies on the polling to detect when a task
+                starts running so you are likely to want to configure more
+                frequent polling.
```
Comment (hjoliver) [inline cylc/flow/cfgspec/globalcfg.py:1036]:
```suggestion
                Note that if the polling
                :cylc:conf:`global.cylc[platforms][<platform name>]
                communication method`
                is used then Cylc relies on polling to detect all task state changes,
                so you may want to configure more
                frequent polling.
```

File: cylc/flow/cfgspec/globalcfg.py [hunk 7]
```diff
@@ -971,59 +972,84 @@
...
-                The intervals between polling after a task job (submitted to
-                the relevant job runner on the relevant host) exceeds its
-                execution time limit. The default setting is PT1M, PT2M, PT7M.
-                The accumulated times (in minutes) for these intervals will be
-                roughly 1, 1 + 2 = 3 and 1 + 2 + 7 = 10 after a task job
-                exceeds its execution time limit.
+                After a task job exceeds its execution time limit Cylc can poll
+                more frequently in the expectation that the task should have
+                completed. A list of interval values can be specified, with the
+                last value used repeatedly until the task reports as finished.
```
Comment (hjoliver) [inline cylc/flow/cfgspec/globalcfg.py:1047]:
```suggestion
                If a job exceeds its execution time limit, Cylc can poll
                more frequently to detect the expected job completion quickly.
                The last interval in the list is used repeatedly until the job completes.
```

**Review by hjoliver — CHANGES_REQUESTED** [review]

File: cylc/flow/cfgspec/globalcfg.py [hunk 8]
```diff
@@ -939,6 +934,10 @@
                  * ``background``
                  * ``slurm``
                  *  ``pbs``
+
+                .. seealso::
+
+                   :ref:`List of built-in Job Runners <tutorial-job-runner>`
```
Comment (dpmatthews) [inline cylc/flow/cfgspec/globalcfg.py:940]: `<AvailableMethods>` is a better reference? (although could do with improving)

Comment (wxtim) [reply cylc/flow/cfgspec/globalcfg.py:940]:
```suggestion
                    :ref:`List of built-in Job Runners <AvailableMethods>`
```

File: cylc/flow/cfgspec/globalcfg.py (hunk 3 above)
Comment (wxtim) [reply cylc/flow/cfgspec/globalcfg.py:1006]:
```suggestion
                A list of ISO 8601 durations/intervals which define when
                the scheduler will resubmit failed jobs.
```

File: cylc/flow/cfgspec/globalcfg.py (hunk 4 above)
Comment (wxtim) [reply cylc/flow/cfgspec/globalcfg.py:1011]:
```suggestion
                The scheduler makes ``$CYLC_TASK_TRY_NUMBER``
                available in the job execution environment. This allows you
                to vary task behavior between submission attempts.
```

File: cylc/flow/cfgspec/globalcfg.py (hunk 3 above)
Comment (hjoliver) [reply cylc/flow/cfgspec/globalcfg.py:1006]: Yes that's fine.

  I have a stylistic dislike of repeating the subject immediately after it was already stated (when, I think, the subject is clear from the context).

Some back up from an academic English style guide:

https://owl.purdue.edu/owl/general_writing/academic_writing/sentence_variety/for_repeated_subjects_or_topics.html

(As a general rule, I also don't like passive voice in technical writing. It tends to be unnecessarily verbose and sounds kind of flaccid).

File: cylc/flow/cfgspec/globalcfg.py (hunk 4 above)
Comment (hjoliver) [reply cylc/flow/cfgspec/globalcfg.py:1011]: Users need to know the meaning of try number (and submit number) and when exactly it increments. So I wouldn't say that's only the "inner workings" of Cylc.

File: cylc/flow/cfgspec/globalcfg.py (hunk 3 above)
Comment (wxtim) [reply cylc/flow/cfgspec/globalcfg.py:1006]: > As a general rule, I also don't like passive voice in technical writing. It tends to be unnecessarily verbose and sounds flaccid. (This is to explain many of my doc PR review suggestions, not yours in particular!)

Agreed, even if I am a culprit.

File: cylc/flow/cfgspec/globalcfg.py (hunk 4 above)
Comment (wxtim) [reply cylc/flow/cfgspec/globalcfg.py:1011]: OK. Fair enough.

File: cylc/flow/cfgspec/globalcfg.py (hunk 7 above)
Comment (dpmatthews) [reply cylc/flow/cfgspec/globalcfg.py:1047]: Need to apply this change in 3 places

File: cylc/flow/cfgspec/globalcfg.py (hunk 3 above)
Comment (wxtim) [reply cylc/flow/cfgspec/globalcfg.py:1006]: N.B - Talking to Dave made this actually about submission and _not_ execution retries.

File: cylc/flow/cfgspec/globalcfg.py (hunk 4 above)
Comment (wxtim) [reply cylc/flow/cfgspec/globalcfg.py:1011]: CYLC_TASK_TRY_NUMBER refers to execution - Remove!

Comment (wxtim) [reply cylc/flow/cfgspec/globalcfg.py:1011]:
```suggestion
```

File: cylc/flow/cfgspec/globalcfg.py (hunk 7 above)
Comment (wxtim) [reply cylc/flow/cfgspec/globalcfg.py]: > Need to apply this change in 3 places

Where else? I can't find this text anywhere else.

**Comment by wxtim** [comment]
Plenty of scope for mission creep here, which I've documented in #4851

File: cylc/flow/cfgspec/globalcfg.py (hunk 7 above)
Comment (dpmatthews) [reply cylc/flow/cfgspec/globalcfg.py]: Sorry - just the "A list of interval values can be specified" bit

File: cylc/flow/cfgspec/globalcfg.py (hunk 4 above)
Comment (hjoliver) [reply cylc/flow/cfgspec/globalcfg.py]: Ohh, good point!!

File: cylc/flow/cfgspec/workflow.py [hunk 9]
```diff
@@ -1118,13 +1117,40 @@ def get_script_common_text(this: str, example: Optional[str] = None):
...
-                desc=DEPRECATED_IN_FAVOUR_OF_PLATFORMS
+                desc='''
+                    List of intervals at which to poll status of job execution.
+
+                    Cylc can poll running jobs to catch problems that prevent
+                    task messages from being sent back to the workflow, such
+                    as hard job kills, network outages, or unplanned job
+                    host shutdown.
+
+                    The lasst interval in the list is used repeatedly until
```
Comment (hjoliver) [inline cylc/flow/cfgspec/workflow.py:1134]:
```suggestion
                    The last interval in the list is used repeatedly until
```

**Review by hjoliver — APPROVED** [review]
:+1: good!

**Comment by hjoliver** [comment]
(Sorry, one conflict now!)

**Pull request merged**